### PR TITLE
Add clearer OpenHands Configuration logs

### PR DESCRIPTION
**Pull request by mamoodi**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Add clearer OpenHands Configuration logs

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

* Add a OpenHands Configuration Logs to clearly show what is being used
* Method docs for people who know more to fill them out when they come across it ;) 

---
**Link of any specific issues this addresses**

### CODE REVIEW

File: openhands/server/session/agent.py [hunk 1]
```diff
@@ -105,16 +132,31 @@ async def _create_controller(
...
 
-        logger.info(f'Agents: {agent_configs}')
-        logger.info(f'Creating agent {agent.name} using LLM {agent.llm.config.model}')
+        logger.info(
+            '\n--------------------------------- OpenHands Configuration ---------------------------------\n'
+            f'LLM: {agent.llm.config.model}\n'
+            f'Base URL: {agent.llm.config.base_url}\n'
+            f'Agent: {agent.name}\n'
+            '-------------------------------------------------------------------------------------------'
+        )
```
Comment (mamoodi) [inline openhands/server/session/agent.py:159]: The "-" may be excessive but the log is too busy and I believe this is one of the most important things we log so it's important that it's visible.

**Review by tobitege — APPROVED** [review]
LGTM

File: openhands/server/session/agent.py [hunk 2]
```diff
@@ -41,11 +47,18 @@ async def start(
...
-        Args:
-            start_event: The start event data (optional).
+        Parameters:
+        - runtime_name (str): The name of the runtime associated with the session
+        - config (AppConfig):
+        - agent (Agent):
+        - max_interations (int):
+        - max_budget_per_task (float):
+        - agent_to_llm_config (dict[str, LLMConfig]):
+        - agent_configs (dict[str, AgentConfig]):
```
Comment (enyst) [inline openhands/server/session/agent.py:59]: FWIW we're not consistent, but we have kind of settled or tried to not include the args types in docstrings. They're visible in hints, and prone to become outdated in docstrings.

Comment (mamoodi) [reply openhands/server/session/agent.py]: Removed types.

File: openhands/server/session/agent.py (hunk 1 above)
Comment (mamoodi) [reply openhands/server/session/agent.py:159]: It's fine for now since it adds a new line by itself. Looks pretty clean

**Pull request merged**